src/legacy/components/TransactionConfirmationModal/LegacyConfirmationPendingContent.tsx

```diff
@@ -0,0 +1,251 @@
+  ApproveWrapper,
+  CloseIconWrapper,
+  CompareItem,
+  ItemList,
+  LowerSection,
+  StepsIconWrapper,
+  StepsWrapper,
+  UpperSection,
+  WalletIcon,
+  Wrapper,
+} from './styled'
+import { ConfirmOperationType } from './types'
+
+enum WalletType {
+  SAFE,
+  SC,
+  EOA,
+}
+
+export function getOperationMessage(operationType: ConfirmOperationType, chainId: number): string {
+  const { native, wrapped } = getChainCurrencySymbols(chainId)
+
+  switch (operationType) {
+    case ConfirmOperationType.WRAP_ETHER:
+      return 'Wrapping ' + native
+    case ConfirmOperationType.UNWRAP_WETH:
+      return 'Unwrapping ' + wrapped
+    case ConfirmOperationType.APPROVE_TOKEN:
+      return 'Approving token'
+    case ConfirmOperationType.ORDER_CANCEL:
+      return 'Canceling your order'
+    case ConfirmOperationType.REVOKE_APPROVE_TOKEN:
+      return 'Revoking token approval'
+    case ConfirmOperationType.CONVERT_VCOW:
+      return 'Converting vCOW to COW'
+    case ConfirmOperationType.CLAIM_VESTED_COW:
+      return 'Claiming vested COW'
+    default:
+      return 'Almost there!'
+  }
+}
+
+function getOperationLabel(operationType: ConfirmOperationType): string {
+  switch (operationType) {
+    case ConfirmOperationType.WRAP_ETHER:
+      return t`wrapping`
+    case ConfirmOperationType.UNWRAP_WETH:
+      return t`unwrapping`
+    case ConfirmOperationType.APPROVE_TOKEN:
+      return t`token approval`
+    case ConfirmOperationType.REVOKE_APPROVE_TOKEN:
+      return t`revoking token approval`
+    case ConfirmOperationType.ORDER_SIGN:
+      return t`order`
+    case ConfirmOperationType.ORDER_CANCEL:
+      return t`cancellation`
+    case ConfirmOperationType.CONVERT_VCOW:
+      return t`vCOW conversion`
+    case ConfirmOperationType.CLAIM_VESTED_COW:
+      return t`vested COW claim`
+  }
+}
+
+function getSubmittedMessage(operationLabel: string, operationType: ConfirmOperationType): string {
+  switch (operationType) {
+    case ConfirmOperationType.ORDER_SIGN:
+      return t`The order is submitted and ready to be settled.`
+    default:
+      return `The ${operationLabel} is submitted.`
+  }
+}
+
+function getWalletNameLabel(walletType: WalletType): string {
+  switch (walletType) {
+    case WalletType.SAFE:
+      return 'Safe'
+    case WalletType.SC:
+      return 'smart contract wallet'
+    case WalletType.EOA:
+      return 'wallet'
+  }
+}
+
+// TODO: replace by common/pure/ConfirmationPendingContent
+export function LegacyConfirmationPendingContent({
+  onDismiss,
+  pendingText,
+  operationType,
+  chainId,
+}: {
+  onDismiss: () => void
+  pendingText: ReactNode
+  operationType: ConfirmOperationType
+  chainId: number
+}) {
+  const { connector } = useWeb3React()
+  const { account } = useWalletInfo()
+  const walletDetails = useWalletDetails()
+  const { ensName, isSmartContractWallet } = walletDetails
+  const gnosisSafeInfo = useGnosisSafeInfo()
+
+  const walletType = useMemo((): WalletType => {
```

Comment:
This could be a separate hook

Reply:
Agree, I would love to refactor it after TWAP launch.
Anyway, in this PR I've not changed the code, just split it into separate files.